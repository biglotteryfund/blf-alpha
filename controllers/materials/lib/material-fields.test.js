/* eslint-env jest */
'use strict';

const { validate } = require('./material-fields');

function mapMessages(validationResult) {
    return validationResult.error.details.map(detail => detail.message);
}

test('should validate material fields schema', () => {
    const invalidResult = validate({
        yourEmail: 'notanemail@invalid'
    });

    expect(mapMessages(invalidResult)).toEqual([
        '"yourName" is required',
        '"yourEmail" must be a valid email',
        '"yourAddress1" is required',
        '"yourTown" is required',
        '"yourCountry" is required',
        '"yourPostcode" is required'
    ]);

    const validResult = validate({
        yourName: 'Björk Guðmundsdóttir',
        yourEmail: 'example@example.com',
        yourAddress1: '3 Embassy Drive',
        yourAddress2: '',
        yourTown: 'Birmingham',
        yourCounty: 'West Midlands',
        yourCountry: 'England',
        yourPostcode: 'B15 1TR',
        yourProjectName: 'Example project name',
        yourGrantAmount: 'under10k',
        yourGrantAmountOther: '',
        yourReason: 'other',
        yourReasonOther: 'Other reason'
    });

    expect(validResult.error).toBeUndefined();
});
