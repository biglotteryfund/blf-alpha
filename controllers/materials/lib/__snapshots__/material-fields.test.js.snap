// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should validate material fields schema 1`] = `
Array [
  "Please provide the 'Name' field",
  "Please provide a valid email address",
  "Please provide the 'Address line 1' field",
  "Please provide the 'Town/city' field",
  "Please provide the 'Country' field",
  "Please provide the 'Postcode' field",
]
`;
